### PR TITLE
Fix deprecations

### PR DESCRIPTION
**Pull request by pierredup**
## Subject

Sonata CoreBundle includes various deprecations in version 3 which will be removed in version 4.
This removes the usage of all the deprecated classes and interfaces for forward-compatibility.

I am targeting this branch, because it's bugfix against the current stable version.

## Changelog

```markdown
### Fixed
- removed usage of deprecated classes and interfaces
```

### CODE REVIEW

File: src/BasketBundle/Controller/Api/BasketController.php [hunk 1]
```diff
@@ -563,7 +563,7 @@ protected function checkExistingCustomerBasket($customerId)
     {
         $basket = $this->basketManager->findOneBy(['customer' => $customerId]);
         if ($basket instanceof BasketInterface) {
-            throw new HttpException('400', sprintf('Customer (%d) already has a basket', $customerId));
+            throw new HttpException(400, sprintf('Customer (%d) already has a basket', $customerId));
```
Comment (greg0ire) [inline src/BasketBundle/Controller/Api/BasketController.php]: Please move this to a separate commit, it does not have to do with the core bundle, same for any unrelated change below.

Comment (greg0ire) [reply src/BasketBundle/Controller/Api/BasketController.php]: I did it for you, and actually made 4 commits.

File: src/CustomerBundle/Twig/Extension/AddressExtension.php [hunk 2]
```diff
@@ -81,7 +80,7 @@ public function renderAddress(\Twig_Environment $environment, $address, $showNam
         $requiredAddressKeys = ['firstname', 'lastname', 'address1', 'postcode', 'city', 'country_code'];
 
         if (!($address instanceof AddressInterface) && (!\is_array($address) || 0 !== \count(array_diff($requiredAddressKeys, array_keys($address))))) {
-            throw new InvalidParameterException(sprintf(
+            throw new \RuntimeException(sprintf(
```
Comment (greg0ire) [inline src/CustomerBundle/Twig/Extension/AddressExtension.php]: @core23 , can you confirm `InvalidParameterException` is deprecated in favor of `\RuntimeException`?

**Comment by greg0ire** [comment]
After this one I will do a release because the last one is so old :sweat_smile:

File: src/CustomerBundle/Twig/Extension/AddressExtension.php (hunk 2 above)
Comment (core23) [reply src/CustomerBundle/Twig/Extension/AddressExtension.php]: Yes https://github.com/sonata-project/SonataCoreBundle/blob/3.x/src/CoreBundle/Exception/InvalidParameterException.php#L16-L18

**Comment by core23** [comment]
> After this one I will do a release because the last one is so old 😅

Have you tested this project? Not sure if this project is working with symfony 3/4...

**Comment by greg0ire** [comment]
I mean a minor release, not a major one, no need for any tests for that, right?

File: src/CustomerBundle/Twig/Extension/AddressExtension.php (hunk 2 above)
Comment (greg0ire) [reply src/CustomerBundle/Twig/Extension/AddressExtension.php]: It doesn't say in favor of what it is deprecated, but ok!
